Code/Engine/RendererVulkan/Device/Implementation/DeviceVulkan.cpp

```diff
@@ -1344,6 +1341,20 @@ void ezGALDeviceVulkan::DestroyVertexDeclarationPlatform(ezGALVertexDeclaration*
   EZ_DELETE(&m_Allocator, pVertexDeclarationVulkan);
 }
 
+void ezGALDeviceVulkan::UpdateBufferForNextFramePlatform(const ezGALBuffer* pBuffer, ezConstByteArrayPtr sourceData, ezUInt32 uiDestOffset)
+{
+  const ezGALBufferVulkan* pBufferVulkan = static_cast<const ezGALBufferVulkan*>(pBuffer);
+
+  m_pInitContext->UpdateBuffer(pBufferVulkan, uiDestOffset, sourceData);
```

Comment:
We can't use `m_pInitContext` for this purpose as this instance is flushed on every submit call, not the start of the frame. This is needed because all resource creation end up in this context and we expect to be able to create resources or block on load in the render code and immediately make use of them.
Thus, in order to ensure these actually end up in the next frame, we need to create a separate instance that is only flushed once at the start of the frame.

Reply:
This is a bit troublesome as we have the following dependencies when creating a resource in the game loop and calling `UpdateBufferForNextFrame`:
1. The resource creation ends up in the `m_pInitContext`.
2. The `UpdateBufferForNextFrame` needs to go in a separate context.
At the start of the frame, before we enter the parallel phase, we need to make one submit call with the command buffer from the `m_pInitContext` followed by the command buffer with all the `UpdateBufferForNextFrame` calls so that resource creation happens before updates.